File: April-18/08-04-18/python/01-Gas-Stations.py
n, x = map(int, input().split())
ar = [int(x) for x in input().split()]

ans = 0
for i in range(0, n):
    if(i > 0):
        ar[i] += ar[i-1]
    if(ar[i] >= x):
        ans = i+1
        break
print(ans)

File: April-18/08-04-18/java/01-Gas-Stations.java

import java.util.*;

class TestClass {
    public static void main(String args[] ) throws Exception {
        Scanner input = new Scanner(System.in);
        int n; long x;
        n = input.nextInt();
        x = input.nextLong();
        long ar[] = new long[n+1];
        for(int i = 1; i <= n; i++)
        {
            ar[i] = input.nextLong();
            if(i > 1)
                ar[i] += ar[i-1];
        }
        int ans = 0;
        for(int i = 1; i <= n; i++)
        {
            if(ar[i] >= x)
            {
                ans = i;
                break;
            }
        }
        System.out.println(ans);
        
    }
}
